### ISSUE TITLE
Admin Credentials to set Global Variables

### ISSUE THREAD

**Issue by Krillle** [ilcato/homebridge-Fibaro-HC2#204]
While it is helpful to use an explicit Fibaro user for Homebridge's read access to select a subset of devices, this no-admin user can't set the as well very helpful global variables acting as a bistable switch. Therefore it would be nice to add separate user credentials for this task:

```
"switchglobalvariables": "PUT A COMMA SEPARATED LIST OF HOME CENTER GLOBAL VARIABLES ACTING LIKE A BISTABLE SWITCH",
"adminUsername": "PUT ADMIN USERNAME OF YOUR HC2 HERE TO SET GLOBAL VARIABLES",
"adminPassword": "PUT ADMIN PASSWORD OF YOUR HC2 HERE TO SET GLOBAL VARIABLES",
```

Changes could be like that: https://github.com/ilcato/homebridge-Fibaro-HC2/pull/205

**Comment by crissmil** [comment]
Great idea! If only it could be implemented...

**Comment by Krillle** [comment]
It’s implemented in my pull request #205

**Comment by crissmil** [comment]
Found it thanks! 
By the way, did u manage to make it work with latest homebridge version 1.3.0?

**Comment by crissmil** [comment]
I've tried it but keep getting this error when I try to modify the switch from iPhone:
"There was a problem setting variable: XXX to true"
If I change the variable value from HC2 it is working on homekit but from homekit to HC2 not.
homebridge 1.3.0
any ideea?

**Comment by Krillle** [comment]
@crissmil The error looks like it's still not using the admin login. Did you add these lines in config.json?

```
"adminUsername": "PUT ADMIN USERNAME OF YOUR HC2 HERE TO SET GLOBAL VARIABLES",
"adminPassword": "PUT ADMIN PASSWORD OF YOUR HC2 HERE TO SET GLOBAL VARIABLES",
```

**Comment by crissmil** [comment]
yes I did. I am using same credentials for changing variables from raspberry python apps and are working...

**Comment by ilcato** [comment]
@crissmil, @Krillle does it works on 1.3? My I merge the PR? I cannot test on HC2.

**Comment by Krillle** [comment]
I will test it. 👍🏻

Am 25.02.2021 um 20:20 schrieb ilcato <notifications@github.com>:

﻿

@crissmil<https://github.com/crissmil>, @Krillle<https://github.com/Krillle> does it works on 1.3? My I merge the PR? I cannot test on HC2.

**Comment by Krillle** [comment]
@ilcato I am going to test the it on Homebridge 1.3. I would need some support for that. I sent you a PM on Homebridge discord.

**Comment by ilcato** [comment]
> @ilcato I am going to test the it on Homebridge 1.3. I would need some support for that. I sent you a PM on Homebridge discord.

@Krillle, I'm not able to see your PM.

**Comment by Krillle** [comment]
@ilcato No problem. It's obsolete. I am about to test it in a fresh install.

**Comment by Krillle** [comment]
@ilcato I don't manage to install the plugin from local Git on a fresh install of Homebridge.io image. Do you see what is the problem? 

```
pi@homebridge:~/homebridge-Fibaro-HC2 $ sudo npm install -g
npm ERR! code 2
npm ERR! path /home/pi/homebridge-Fibaro-HC2
npm ERR! command failed
npm ERR! command sh -c tsc
npm ERR! src/fibaro-api.ts(19,26): error TS2307: Cannot find module 'request' or its corresponding type declarations.
npm ERR! src/fibaro-api.ts(36,8): error TS2339: Property 'adminAuth' does not exist on type 'FibaroClient'.
npm ERR! src/fibaro-api.ts(38,29): error TS2339: Property 'adminAuth' does not exist on type 'FibaroClient'.
npm ERR! src/index.ts(46,26): error TS2307: Cannot find module 'request' or its corresponding type declarations.
```

**Comment by Krillle** [comment]
@ilcato I don't manage to install the plugin from my GitHub to a fresh Homebridge.io image install. Do you know how to solve this?

```
pi@homebridge:~/homebridge-Fibaro-HC2 $ sudo npm install -g https://github.com/Krillle/homebridge-Fibaro-HC2.git
npm ERR! code 128
npm ERR! command failed
npm ERR! command git clone ssh://git@github.com/Krillle/homebridge-Fibaro-HC2.git /root/.npm/_cacache/tmp/git-clone-03c45e2c --recurse-submodules --depth=1
npm ERR! /bin/bash: /root/.bashrc: Permission denied
npm ERR! fatal: could not create leading directories of '/root/.npm/_cacache/tmp/git-clone-03c45e2c': Permission denied


root@homebridge:~# npm install -g https://github.com/Krillle/homebridge-Fibaro-HC2.git
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm ERR! code 217
npm ERR! git dep preparation failed
npm ERR! command /usr/local/bin/node /usr/local/lib/node_modules/npm/bin/npm-cli.js install --force --cache=/root/.npm --prefer-offline=false --prefer-online=false --offline=false --no-progress --no-save --no-audit
npm ERR! npm WARN using --force Recommended protections disabled.
npm ERR! npm ERR! code ENOTEMPTY
npm ERR! npm ERR! syscall rename
npm ERR! npm ERR! path /usr/local/lib/node_modules/homebridge-fibaro-hc2
npm ERR! npm ERR! dest /usr/local/lib/node_modules/.homebridge-fibaro-hc2-dImggSuB
npm ERR! npm ERR! errno -39
npm ERR! npm ERR! ENOTEMPTY: directory not empty, rename '/usr/local/lib/node_modules/homebridge-fibaro-hc2' -> '/usr/local/lib/node_modules/.homebridge-fibaro-hc2-dImggSuB'
```

**Comment by Krillle** [comment]
@ilcato The added feature to switch global variables with seperate admin user is working in Homebridge v1.3.1.
Additionally I added  Hombridge UI settings schema. 
Minor change: I found in code pollerperiod default is 5. I corrected this in config.json comments. 

However the new feature seems to break the polling and setting of normal devices. Despite checking and investigating for hours, I could nod find the reason for this. Maybe you will see the reason having have a look on the lines of code I changed. 

It might be a side effect, as I was not able to install the plugin from my local Git. I had to copy the compiled files manually. Would need some support here.
